src/types/mod.rs

```diff
@@ -424,6 +424,64 @@ impl ValueInternal for JsNumber {
     }
 }
 
+/// A JavaScript Date object.
+#[repr(C)]
+#[derive(Clone, Copy)]
+#[cfg(feature = "napi-runtime")]
+pub struct JsDate(raw::Local);
+
+#[cfg(feature = "napi-runtime")]
+impl Value for JsDate { }
+
+#[cfg(feature = "napi-runtime")]
+impl Managed for JsDate {
+    fn to_raw(self) -> raw::Local { self.0 }
+
+    fn from_raw(_: Env, h: raw::Local) -> Self { JsDate(h) }
+}
+
+#[cfg(feature = "napi-runtime")]
+impl JsDate {
+    pub const MIN_VALID_VALUE: f64 = -8640000000000000.0;
+    pub const MAX_VALID_VALUE: f64 = 8640000000000000.0;
+
+    pub(crate) fn new_internal<'a>(env: Env, value: f64) -> Handle<'a, JsDate> {
+        unsafe {
+            let mut local: raw::Local = std::mem::zeroed();
+            neon_runtime::date::new_date( env.to_raw(), &mut local, value);
+            Handle::new_internal(JsDate(local))
+        }
+    }
+
+    pub fn new<'a, C: Context<'a>, T: Into<f64>>(cx: &mut C, time: T) -> Handle<'a, JsDate> {
+        JsDate::new_internal(cx.env(), time.into())
+    }
+
+    pub fn value<'a, C: Context<'a>>(self, cx: &mut C) -> f64 {
+        let env = cx.env().to_raw();
+        unsafe {
+            neon_runtime::date::value(env, self.to_raw())
+        }
+    }
+
+    pub fn is_valid<'a, C: Context<'a>>(self, cx: &mut C) -> bool {
+        let value = self.value(cx);
+        return value <= JsDate::MAX_VALID_VALUE && value >= JsDate::MIN_VALID_VALUE;
```

Comment:
Does n-api really return the original value here? In Js it returns `NaN`. Is this documented behavior?

Reply:
`self.value` returns `std::f64::NAN` if >= 2**53. Will add test case for this.